tests/test_project/tests/test_qa.py

```diff
@@ -189,5 +207,23 @@ def test_qa_call_check_commit_message_failure(self):
                 else:
                     self.fail('SystemExit or Exception not raised')
 
+    def test_commit_failure_message(self):
+        bad_commit = [
+            'commitcheck',
+            '--message',
+            "[qa] Updated file and fixed problem\n\n"
+            "Added more files. Fixes #20"
+        ]
+        captured_output = io.StringIO()
+        sys.stdout = captured_output  # redirect stdout
+        with patch('argparse._sys.argv', bad_commit):
+            try:
+                check_commit_message()
+            except (SystemExit):
+                pass
+        sys.stdout = sys.__stdout__  # reset redirect
```

Comment:
Do you need to do this even if you are inside the `patch`'s scope?

Reply:
yes, even if I will move `sys.stdout = captured_output` to `patch` scope I will have to anyway reset `sys.stdout`